src/redist/targets/packaging/osx/clisdk/Distribution-Template-x64

```diff
@@ -0,0 +1,90 @@
+    </choice>
+    <choice id="{NetCoreAppTargetingPackComponentId}.pkg.arm64" visible="true" selected="!IsX64Machine()" customLocation="{x64EmulationPkgInstallDirectory}" title="{NetCoreAppTargetingPackBrandName} ({arch})" description="The .NET Targeting Pack">
+        <pkg-ref id="{NetCoreAppTargetingPackComponentId}.pkg" />
+    </choice>
+    <choice id="{NetStandardTargetingPackComponentId}.pkg.arm64" visible="true" selected="!IsX64Machine()" customLocation="{x64EmulationPkgInstallDirectory}" title="{NetStandardTargetingPackBrandName} ({arch})" description="The .NET Standard 2.1 Targeting Pack">
+        <pkg-ref id="{NetStandardTargetingPackComponentId}.pkg" />
+    </choice>
+    <choice id="{NetCoreAppHostPackComponentId}.pkg.arm64" visible="true" selected="!IsX64Machine()" customLocation="{x64EmulationPkgInstallDirectory}" title="{NetCoreAppHostPackBrandName} ({arch})" description="The .NET App Host Pack">
+        <pkg-ref id="{NetCoreAppHostPackComponentId}.pkg" />
+    </choice>
+    <choice id="{SharedFxComponentId}.pkg.arm64" visible="true" selected="!IsX64Machine()" customLocation="{x64EmulationPkgInstallDirectory}" title="{SharedFxBrandName} ({arch})" description="The .NET Shared Framework">
+        <pkg-ref id="{SharedFxComponentId}.pkg" />
+    </choice>
+    <choice id="{HostFxrComponentId}.pkg.arm64" visible="true" selected="!IsX64Machine()" customLocation="{x64EmulationPkgInstallDirectory}" title="{HostFxrBrandName} ({arch})" description="The .NET Host FX Resolver">
+        <pkg-ref id="{HostFxrComponentId}.pkg" />
+    </choice>
+    <choice id="{SharedHostComponentId}.pkg.arm64" visible="true" selected="!IsX64Machine()" customLocation="{x64EmulationPkgInstallDirectory}" title="{SharedHostBrandName} ({arch})" description="The .NET Shared Host.">
+        <pkg-ref id="{SharedHostComponentId}.pkg" />
+    </choice>
+    <choice id="{CLISdkComponentId}.pkg.arm64" visible="true" selected="!IsX64Machine()" customLocation="{x64EmulationPkgInstallDirectory}" title="{CLISdkBrandName} ({arch})" description="The .NET SDK">
+        <pkg-ref id="{CLISdkComponentId}.pkg"/>
+    </choice>
+    <pkg-ref id="{NetCoreAppTargetingPackComponentId}.pkg">{NetCoreAppTargetingPackComponentId}.pkg</pkg-ref>
+    <pkg-ref id="{NetStandardTargetingPackComponentId}.pkg">{NetStandardTargetingPackComponentId}.pkg</pkg-ref>
+    <pkg-ref id="{NetCoreAppHostPackComponentId}.pkg">{NetCoreAppHostPackComponentId}.pkg</pkg-ref>
+    <pkg-ref id="{SharedFxComponentId}.pkg">{SharedFxComponentId}.pkg</pkg-ref>
+    <pkg-ref id="{HostFxrComponentId}.pkg">{HostFxrComponentId}.pkg</pkg-ref>
+    <pkg-ref id="{SharedHostComponentId}.pkg">{SharedHostComponentId}.pkg</pkg-ref>
+    <pkg-ref id="{CLISdkComponentId}.pkg">{CLISdkComponentId}.pkg</pkg-ref>
+    <script>
+<![CDATA[
+function IsX64Machine() {
+    var machine = system.sysctl("hw.machine");
+    system.log("Machine type: " + machine);
+    
+    var result = machine == "x64" || machine.endsWith("_x64");
```

Comment:
Can you double check this on an x64 machine.  I just noticed that @am11 [pointed out](https://github.com/dotnet/runtime/pull/59210#discussion_r710746019) I had the uname value incorrect, this could be wrong too.  I only tested the initial prototype on an M1.  If this is reflecting the `uname -m` value then we have a bug here (and in the Arcade commit).

Reply:
Yep, confirmed with a x64 machine that this should be `_64`, updated here as well as in arcade with https://github.com/dotnet/arcade/pull/7913

Reply:
This looks like JavaScript. :)
I couldn't find this `system` module in node.js which has `sysctl`, but in shell on BigSur, `sysctl -a | grep machine` does not return anything (i.e. `hw.machine` key does not exist). Some related entries are:

```
kern.version: Darwin Kernel Version 20.6.0: Wed Jun 23 00:26:31 PDT 2021; root:xnu-7195.141.2~5/RELEASE_X86_64
hw.optional.x86_64: 1
```

Reply:
It's not listed in -a but you can fetch it.  Try running `sysctl hw.machine`.  Yes this is javascript, this is documented here: https://developer.apple.com/documentation/installer_js
https://developer.apple.com/documentation/installer_js/system/1812308-sysctl

Reply:
Looking at other `syctl` values, I wonder if there is something here that might be more stable / documented.  I see `hw.cputype` which seems to match the values defined in [machine.h](https://opensource.apple.com/source/xnu/xnu-4570.41.2/osfmk/mach/machine.h.auto.html).

```
#define CPU_ARCH_ABI64	0x01000000		/* 64 bit ABI */
#define CPU_TYPE_X86		((cpu_type_t) 7)
#define CPU_TYPE_ARM		((cpu_type_t) 12)
#define CPU_TYPE_X86_64		(CPU_TYPE_X86 | CPU_ARCH_ABI64)
#define CPU_TYPE_ARM64          (CPU_TYPE_ARM | CPU_ARCH_ABI64)
```

On my M1  the value is 16777228 (0x0100000C)

Which seems does correspond to `CPU_TYPE_ARM64`.  I would expect x86_64 to return 0x01000007 = 16777223

What do folks think about using this instead?

Reply:
Just found this blog which discusses this: https://ohanaware.com/blog/202032/macOS-CPU-Architecture.html

Apple has [this page](https://developer.apple.com/documentation/apple-silicon/addressing-architectural-differences-in-your-macos-code), which suggests using sysctl to fetch stuff and seems to give carte blanche to whatever it dumps with `sysctl -a`